### DISCUSSION TITLE
Return error when attempting to revert a migration if no down block is defined

### DISCUSSION THREAD

**Discussion by benjamin-thomas** [amacneil/dbmate#363] [Feature Ideas]
v1.10.0

Hello,

Is there a way to ensure no rollback ever occurs. I tried removing the `migrate:down` keyword but rolling back then succeeds, which I don't want

For the moment, this seems to do the trick. Do you see a better way?

```sql
-- migrate:up
CREATE ROLE whatever;

-- migrate:down
SELECT 1/0
```

**Comment by amacneil** [comment]
What are you trying to achieve? If you don't define a down migration (or add a no-op statement as you propose) then I believe dbmate will still remove record of that migration but will not make any changes to the database.

**Comment by benjamin-thomas** [comment]
Yes that's the problem :)

If there is no `migrate:down` instruction, it'd be useful if that record wouldn't be removed.

Otherwise the migration is marked "pending" via the CLI and will be ran once more on `dbmate migrate`, but then fails as it was meant to be ran once.

I gave an example of a non idempotent command but that migration could be anything, cleanup data, changing a db design that should never be reversed, etc.. 

Using a division by zero does the trick for me, I just wondered if that was something handled and not documented.

**Comment by amacneil** [comment]
Let's keep this issue open, I think it would be nice to explicitly fail when attempting to rollback a migration if no down migration is defined, rather than pretending that something happened.

Of course, it's still entirely up to the developer to ensure that a down migration is the inverse of an up migration, and/or that the up migration is idempotent if desired.

**Comment by gregwebs** [comment]
I have a PR #434 where you can specify specific migrations to do or to rollback: #434. It seems like that would solve this issue?
